### PR TITLE
[admission-policy-engine] Add ability to disable service account auto mount

### PR DESCRIPTION
**Pull request by Dmitrykob**
## Description

Added a security policy to control the mounting of service account tokens in pods.
Security policy flag "automountServiceAccountToken [true/false]" defines whether the use of automatic mounting of service account token is allowed.
```
---
apiVersion: deckhouse.io/v1alpha1
kind: SecurityPolicy
metadata:
  name: mypolicy
spec:
  enforcementAction: Deny
  policies:
    automountServiceAccountToken: true
  match:
    namespaceSelector:
      labelSelector:
        matchLabels:
          enforce: mypolicy
```

By default, automatic mounting of a service account token is disallowed when the flag is false or absent.

## Why do we need it, and what problem does it solve?

fix https://github.com/deckhouse/deckhouse/issues/5637

## What is the expected result?

Test case (with "automountServiceAccountToken": "false"): 

```
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  containers:
  - name: nginx
    image: nginx:1.14.2
    ports:
    - containerPort: 80
    securityContext:
      readOnlyRootFilesystem: false
      allowPrivilegeEscalation: false
```

```
kubectl apply -n test-namespace -f test-pod.yaml
Error from server (Forbidden): error when creating "test-pod.yaml": admission webhook "admission-policy-engine.deckhouse.io" denied the request: [mypolicy] Automounting service account token is disallowed, pod: test
```

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: admission-policy-engine
type: fix
summary: Add a security policy to control the mounting of service account tokens in pods. This rule will prohibit running pods with automountServiceAccountToken for existing SecuirtyPolicy. To restore the old behavior, add `automountServiceAccountToken: true` to the policy.
impact_level: high
```

### CODE REVIEW

**Review by z9r5 — APPROVED** [review]

File: modules/015-admission-policy-engine/crds/doc-ru-security-policy.yaml [hunk 1]
```diff
@@ -141,6 +141,8 @@ spec:
                                 description: "Максимальное значение диапазона."
                     readOnlyRootFilesystem:
                       description: "Определяет необходимость запуска контейнеров с корневой файловой системой, доступной только для чтения"
+                    automountServiceAccountToken:
+                      description: "Определяет возможность для подов включать автоматическое монтирование токена сервисного аккаунта"
```
Comment (yalosev) [inline modules/015-admission-policy-engine/crds/doc-ru-security-policy.yaml:145]:
```suggestion
                      description: "Определяет возможность запуска подов с автоматически примонтированным токеном сервисного аккаунта"
```

File: modules/015-admission-policy-engine/crds/security-policy.yaml [hunk 2]
```diff
@@ -339,6 +339,9 @@ spec:
                     readOnlyRootFilesystem:
                       type: boolean
                       description: "Defines if it's possible to run containers with non-read only file system."
+                    automountServiceAccountToken:
+                      type: boolean
+                      description: "Defines if it's possible to run containers with enabled automountServiceAccountToken."
```
Comment (yalosev) [inline modules/015-admission-policy-engine/crds/security-policy.yaml:344]:
```suggestion
                      description: "Defines if it's possible to run pods with enabled automountServiceAccountToken."
```

**Review by yalosev — APPROVED** [review]

File: modules/015-admission-policy-engine/crds/doc-ru-security-policy.yaml [hunk 3]
```diff
@@ -141,6 +141,8 @@ spec:
...
                     seccompProfiles:
                       description: "Определяет список доступных к использованию seccomp-профилей."
```
Comment (MariaMakeeva) [inline modules/015-admission-policy-engine/crds/doc-ru-security-policy.yaml:147]:
```suggestion
                      description: Определяет список доступных seccomp-профилей.
```

**Review by yalosev — CHANGES_REQUESTED** [review]
Have to add a migration for existing SecurityPolicy here

**Review by MariaMakeeva — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

File: modules/015-admission-policy-engine/hooks/handle_security_policies.go [hunk 4]
```diff
@@ -110,6 +110,10 @@ func (sp *securityPolicy) preprocesSecurityPolicy() {
 			sp.Spec.Policies.SupplementalGroups = nil
 		}
 	}
+	if sp.Spec.Policies.AutomountServiceAccountToken == nil {
+		sp.Spec.Policies.AutomountServiceAccountToken = newTrue()
```
Comment (yalosev) [inline modules/015-admission-policy-engine/hooks/handle_security_policies.go:114]:
```suggestion
		sp.Spec.Policies.AutomountServiceAccountToken = pointer.Bool(true)
```

**Pull request merged**